examples/twittermap/web/public/javascripts/sidebar/controllers.js

```diff
@@ -170,17 +172,124 @@ angular.module("cloudberry.sidebar", ["cloudberry.common"])
         $scope.hashTagsList = newResult;
       }
     );
+
+    // return difference of two arrays, the arrays must has no duplicate
+    function arrayDiff (newArray, oldArray) {
```

Comment:
Can we extract these two functions (`arrayDiff` and `preProcess`) as common functions since they also appear in popup window PR: https://github.com/ISG-ICS/cloudberry/pull/580/files

Reply:
Sure. Where should I add these function? Should I create a new `folder` and create a `service.js` in `cloudberry/examples/twittermap/web/public/javascripts`?

Reply:
maybe a new `chart-util.js` file in `javacripts/common`?

Reply:
OK.